src/app/github-link-area/github-link-area.component.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  Component,
+  Input,
+  ViewEncapsulation,
+  OnChanges,
+  SimpleChanges,
+  ViewChild,
+  ElementRef
+} from '@angular/core';
+
+import { GitHubLinkService } from './github-link.service';
+
+/**
+ * GitHub Link Area component. This shows a formatted link to GitHub with an
+ * indicator of whether the GitHub issue is open or closed in an text area.
+ * Works only for public remote repositories for now. See example for how to
+ * use this.
+ */
+@Component({
+  encapsulation: ViewEncapsulation.None,
+  selector: 'github-link-area',
+  styles: [ `
+    .gh-link-open { color: @color-pf-red; }
+    .gh-link-closed { color: @color-pf-green; }
+    .gh-link-error { color: @color-pf-orange; }
+    `],
+  templateUrl: './github-link-area.component.html'
+})
+export class GitHubLinkAreaComponent implements OnChanges {
+
+  @Input('content') content: string;
+
+  constructor(private gitHubLinkService: GitHubLinkService) {}
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes && changes.content) {
+      this.updateOnChanges();
+    }
+  }
+
+  /*
+   * Because Angular testing is broken, we need to be able to call
+   * this from a test. See reason here:
+   * https://github.com/angular/angular/issues/9866
+   */
+  updateOnChanges() {
+    this.updateLinkTexts();
+    this.updateLinks();
+  }
+
+  /*
+   * Replaces the match (which should be the default icon) with an icon that
+   * indicates the status contained in linkData.
+   */
+  replaceLink(linkData: any): void {
+    this.content = this.content.split(linkData.match).join(
+      // tslint:disable-next-line:max-line-length
+      (linkData.state === 'open' ? '<span class="fa fa-clock-o gh-link-open" tooltip="Issue Open"></span>' : '') +
+      (linkData.state === 'closed' ? '<span class="fa fa-check gh-link-closed tooltip="Issue Closed""></span>' : '') +
```

Comment:
Something odd is going on with your double quotes. It looks like you missed on after `gh-link-closed`

Reply:
Yeah, cut-and-paste sucks. Fixed.